queue_job/models/queue_job.py

```diff
@@ -370,6 +381,18 @@ def set_done(self):
         return {'type': 'ir.actions.act_window_close'}
 
 
+class SetJobsToCancelled(models.TransientModel):
+    _inherit = 'queue.requeue.job'
+    _name = 'queue.jobs.to.cancelled'
+    _description = 'Cancel all selected jobs'
+
+    @api.multi
+    def set_cancelled(self):
+        jobs = self.job_ids.filtered(lambda x: x.state in ('pending', 'failed'))
```

Comment:
Can you add that corner case `enqueued` here as it's listed as a valid state for `button_cancelled` below.